array.go

```diff
@@ -33,15 +33,15 @@ func (a array) index(i int, size, offset uintptr) unsafe.Pointer {
 	return unsafe.Add(a.ptr, uintptr(i)*size+offset)
 }
 
-func (a array) slice(i, j int, size, offset uintptr) array {
+func (a array) slice(i, j int, size uintptr) array {
```

Comment:
This was a bug, when slicing we need the subarray to start at the beginning of an element, not at the offset of a field.